tests/Scout/ScoutIntegrationTest.php

```diff
@@ -0,0 +1,242 @@
+        $app['config']->set('scout.prefix', 'prefix_');
+    }
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        // Init the SQL database with some objects that will be indexed
+        // Test data copied from Laravel Scout tests
+        // https://github.com/laravel/scout/blob/10.x/tests/Integration/SearchableTests.php
+        ScoutUser::executeSchema();
+
+        $collect = LazyCollection::make(function () {
+            yield ['name' => 'Laravel Framework'];
+
+            foreach (range(2, 10) as $key) {
+                yield ['name' => 'Example ' . $key];
+            }
+
+            yield ['name' => 'Larry Casper', 'email_verified_at' => null];
+            yield ['name' => 'Reta Larkin'];
+
+            foreach (range(13, 19) as $key) {
+                yield ['name' => 'Example ' . $key];
+            }
+
+            yield ['name' => 'Prof. Larry Prosacco DVM', 'email_verified_at' => null];
+
+            foreach (range(21, 38) as $key) {
+                yield ['name' => 'Example ' . $key, 'email_verified_at' => null];
+            }
+
+            yield ['name' => 'Linkwood Larkin', 'email_verified_at' => null];
+            yield ['name' => 'Otis Larson MD'];
+            yield ['name' => 'Gudrun Larkin'];
+            yield ['name' => 'Dax Larkin'];
+            yield ['name' => 'Dana Larson Sr.'];
+            yield ['name' => 'Amos Larson Sr.'];
+        });
+
+        $id = 0;
+        $date = new DateTimeImmutable('2021-01-01 00:00:00');
+        foreach ($collect as $data) {
+            $data = array_merge(['id' => ++$id, 'email_verified_at' => $date], $data);
+            ScoutUser::create($data)->save();
+        }
+
+        self::assertSame(44, ScoutUser::count());
+    }
+
+    /** This test create the search index for tests performing search */
+    public function testItCanCreateTheCollection()
+    {
+        $this->skipIfSearchIndexManagementIsNotSupported();
+
+        $collection = DB::connection('mongodb')->getCollection('prefix_scout_users');
+        $collection->drop();
+
+        // Recreate the indexes using the artisan commands
+        // Ensure they return a success exit code (0)
+        self::assertSame(0, artisan($this, 'scout:delete-index', ['name' => ScoutUser::class]));
+        self::assertSame(0, artisan($this, 'scout:index', ['name' => ScoutUser::class]));
+        self::assertSame(0, artisan($this, 'scout:import', ['model' => ScoutUser::class]));
+
+        self::assertSame(44, $collection->countDocuments());
+
+        $searchIndexes = $collection->listSearchIndexes(['name' => 'scout']);
+        self::assertCount(1, $searchIndexes);
+
+        // Wait for all documents to be indexed asynchronously
+        $i = 100;
+        while (true) {
+            $indexedDocuments = $collection->aggregate([
+                ['$search' => ['index' => 'scout', 'exists' => ['path' => 'name']]],
+            ])->toArray();
+
+            if (count($indexedDocuments) > 0) {
```

Comment:
Is it an "all or nothing" approach, such that if `$search` returns at least one document we can safely assert that all 44 have been indexed?

If not, should this check for `>= 44`?

Reply:
Updated to `>= 44`, thanks.